framework/src/base/DisplacedSystem.C

```diff
@@ -39,3 +44,39 @@ DisplacedSystem::getVector(std::string name)
     return _undisplaced_system.getVector(name);
 }
 
+NumericVector<Number> &
+DisplacedSystem::addVector(const std::string & vector_name, const bool project, const ParallelType type, bool zero_for_residual)
+{
+  if (hasVector(vector_name))
+    return getVector(vector_name);
+
+  NumericVector<Number> * vec = &_sys.add_vector(vector_name, project, type);
+
+  return *vec;
+}
+
+void
+DisplacedSystem::saveOldSolutions()
+{
+  if (!_sys_sol_old)
+    _sys_sol_old= &addVector("save_flux_old", false, PARALLEL);
+  if (!_aux_sol_old)
+    _aux_sol_old = &_displaced_problem.auxSys().addVector("save_aux_old",  false, PARALLEL);
+  if (!_sys_sol_older)
+    _sys_sol_older = &addVector("save_flux_older", false, PARALLEL);
+  if (!_aux_sol_older)
+    _aux_sol_older = &_displaced_problem.auxSys().addVector("save_aux_older",  false, PARALLEL);
```

Comment:
Actually the old and older solution vectors need to be saved only for the inverse power iteration because the solution of previous iteration is using the old solution vector and the older solution vector. If the power iteration happens within a time step, we do not want the old solutions to be overwritten, so we save them and restore them after power iteration is done. The vector used for saving old solutions can be destroyed after power iteration is done to save memory. Does this make sense?